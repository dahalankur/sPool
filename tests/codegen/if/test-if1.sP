if (true && (false || true)):
    # we are in!
    println("I'm in captain")
    if (false):
        println("Not here though...sigh")
    else println("In here too!");
else
    println("You'll never get here, you'll never get here!") # else of outermost if
;

if (false):
    # empty statement list is allowed!
else
    if (true):
        if (true):
            if (true):
                if (true):
                    println("I'm in too deep")
                    if (false):
                    else println("And I'm tryin to keep")
                    ;
                ;
            ;
        else print("1");
        print(int_to_string(1 + 1)) # will print 2!
    ;
;