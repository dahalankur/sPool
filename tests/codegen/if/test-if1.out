I'm in captain
In here too!
I'm in too deep
And I'm tryin to keep
2